### PR TITLE
42 styling

### PR DESCRIPTION
**Pull request by kanenathan213**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -32,22 +32,30 @@
...
+                <nav class="navbar navbar-default navbar-fixed-top navbar-custom">
+                    <div class="container-fluid">
+                        <div class="navbar-header">
+                            <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target=".navbar-collapse-1">
+                                <span class="sr-only">Toggle navigation</span>
+                                <span class="icon-bar"></span>
+                                <span class="icon-bar"></span>
+                                <span class="icon-bar"></span>
+                              </button>
+                            <a class="header-text navbar-brand" ng-click="goToRoute('/income')">
```
Comment (bobo333) [inline index.html]: I'm sure there's a reason, but why make this an `ng-click` instead of just `href`, considering it's an `<a>` tag?

File: index.html [hunk 2]
```diff
@@ -32,22 +32,30 @@
...
+                                <img src="/img/bowl.png" id="brand-logo"><span class="product-name">Plenti</span>
+                            </a>
+                        </div>
+                        <div class="collapse navbar-collapse navbar-collapse-1">
+                            <ul class="nav navbar-nav navbar-right">
+                                <li><a class="nav-item-custom" ng-click="goToRoute('/about')">About</a></li>
```
Comment (bobo333) [inline index.html]: same with this one?

File: javascript/app.js [hunk 3]
```diff
@@ -39,7 +39,7 @@ FinancialFreedom.directive('autofocus', ['$timeout', function($timeout) {
 }]);
 
 FinancialFreedom.controller('bodyController', ['$location',  function($location) {
-    $location.path("/")
+    $location.path("/");
```
Comment (bobo333) [inline javascript/app.js]: why is this here?

File: javascript/app.js [hunk 4]
```diff
@@ -99,11 +99,46 @@ FinancialFreedom.controller('ExpensesInputController', ['$scope', '$location', '
...
+    }
+});
+
 FinancialFreedom.controller('TimeToRetirementController', ['$scope', 'RetirementCalculatorService', 'CreateRetirementGraphService', function($scope, RetirementCalculatorService, CreateRetirementGraphService) {
     
     var retirement_data = RetirementCalculatorService.calculateRetirementInfo();
+
+    $(function () {
+        $('[data-toggle="popover"]').popover();
+    });
```
Comment (bobo333) [inline javascript/app.js]: why is this wrapped in a jquery function and not simply call `$('[data-toggle="popover"]').popover();` outside?

File: javascript/app.js [hunk 5]
```diff
@@ -99,11 +99,46 @@ FinancialFreedom.controller('ExpensesInputController', ['$scope', '$location', '
...
+        }
+        if (output_value == 'incomeincrease') {
+            $scope.incomeincrease = $scope.incomeincrease + increment;
+        }
+        if (output_value == 'expensesincrease') {
+            $scope.expensesincrease = $scope.expensesincrease + increment;
+        }
+        if (output_value == 'growth') {
+            $scope.growth = $scope.growth + increment;
+        }
```
Comment (bobo333) [inline javascript/app.js]: these should probably all be `else if` instead of separate if statements. Also, there might be a better way where instead of passing in `output_value` as a string, pass in the actual model that's supposed to be incremented. I can't remember if angular can do that nicely, though.

File: partials/about.html [hunk 6]
```diff
@@ -14,10 +14,10 @@
...
                     	<div class="supporting-section-title">About us</div>
                     	<div class="supporting-section-body">
-                        	We are <a href="https://github.com/bobo333" target="_blank">Steve</a>, Pratik, and <a href="http://www.nathanrkane.com" target="_blank">Nathan</a>, the makers of Plenti. We believe that people lead more fulfilling lives in financial security, when they can choose how they spend their time. We hope Plenti sheds light on how attainable financial freedom is for many working people, and inspires the lifestyle adjustments that bring financial independence more quickly, like smart spending and investing.
+                        	We are <a href="https://github.com/bobo333" target="_blank">Steve</a> and <a href="http://www.nathanrkane.com" target="_blank">Nathan</a>, the makers of Plenti. We believe that people lead more fulfilling lives in financial security, when they can choose how they spend their time. We hope Plenti sheds light on how attainable financial freedom is for many working people, and inspires the lifestyle adjustments that bring financial independence more quickly, like smart spending and investing.
```
Comment (bobo333) [inline partials/about.html]: ditching tiki? haha

File: partials/time_to_retirement.html [hunk 7]
```diff
@@ -34,102 +34,105 @@
...
-                        <div id="expenses-control" class="input-control col-md-offset-3 col-md-5">
-                            <div class="">
-                            	<div class="">
-						        	<button class="btn btn-default glyphicon-plus control-increase-button control-change-button" type="button" ng-click="expenses = expenses + 100"></button>
+
+                    <div class="output-control-container col-xs-12 col-md-4">
+                        <div id="expenses-control" class="input-control primary-control col-xs-offset-3 col-xs-6 col-md-offset-3 col-md-6">
+                            <div>
+                            	<div>
+						        	<button class="btn btn-default glyphicon-plus control-increase-button control-change-button" type="button" ng-click="incrementOutputValue('expenses', 100)"></button>
```
Comment (bobo333) [inline partials/time_to_retirement.html]: You might be able to write this as `ng-click="incrementOutputValue(expenses, 100)"` (note the lack of parentheses around `expenses`, meaning it's actually passing the scope variable `expenses` instead of a string `"expenses"`. But it might not let you. If it does, then that big if statement in the controller can be removed.

File: sass/partials/_animations.scss [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+  display:block!important;
+}
+.hide-fade.ng-hide-add.ng-hide-add-active,
+.hide-fade.ng-hide-remove {
+  opacity:0;
+}
+.hide-fade.ng-hide-add,
+.hide-fade.ng-hide-remove.ng-hide-remove-active {
+  opacity:1;
+}
```
Comment (bobo333) [inline sass/partials/_animations.scss]: try to keep indentation consistent, the first couple blocks look like tabs, while the last couple look like spaces.

File: sass/partials/_next_button.scss [hunk 9]
```diff
@@ -1,29 +1,22 @@
 $next-button-height-min-sm: 80px;
 $next-button-height: 80px;
-$next-button-icon-color: $secondary_brown;
-$next-button-border-color: $secondary_brown_dark;
+$next-button-border-color: $secondary_brown;
+$next-button-icon-color: $next-button-border-color;
```
Comment (bobo333) [inline sass/partials/_next_button.scss]: is there a reason to assign one variable to another? why not just use `$next-button-border-color` wherever `$next-button-icon-color is used`?

File: sass/partials/_time_to_retirement.scss [hunk 10]
```diff
@@ -449,5 +519,4 @@ input[type="text"].form-control {
     position: absolute;
     left: 15%;
     right: 15%;
-}
-
+}
```
Comment (bobo333) [inline sass/partials/_time_to_retirement.scss]: 500 lines of css seems like a lot for one view. There might be a way to consolidate some of this into classes that are re-usable throughout this view. Not a huge deal though.

File: index.html (hunk 1 above)
Comment (kanenathan213) [inline index.html]: Nope, no reason. I made them href's instead

File: partials/about.html (hunk 6 above)
Comment (kanenathan213) [inline partials/about.html]: Haha. Gotta commit some code if you want to be a creator.

File: sass/partials/_time_to_retirement.scss (hunk 10 above)
Comment (kanenathan213) [inline sass/partials/_time_to_retirement.scss]: Yeah you're probably right, and also we should break them into multiple partials, since there are pretty distinct sections.

File: sass/partials/_next_button.scss (hunk 9 above)
Comment (kanenathan213) [inline sass/partials/_next_button.scss]: Yeah good call.

File: partials/time_to_retirement.html (hunk 7 above)
Comment (kanenathan213) [inline partials/time_to_retirement.html]: I think you can pass scope variables like that; however, where I'm getting stuck is how to keep the scope updated. I tried $scope.scope_variable = $scope.scope_variable + increment, but that doesn't work with scope_variable being an argument of the method.

Comment (bobo333) [inline partials/time_to_retirement.html]: Ah yes, this is one of the more annoying aspects of angular. For some reason, passing literals (just a number, or string, basically non-objects) breaks the scope connection when passed around like that. I'm not sure why, but there's info online if you're interested. The solution is to put all the values in an object in the scope. What I mean is instead of using `$scope.scope_variable`, use `$scope.scope_object.value`. Then you can pass the whole object, `$scope.scope_object` and increment `$scope.scope_object.value` in the controller and the binding shouldn't be broken.

If that's too confusing or gives you more problems then don't worry about it and just leave it as it was, not a huge deal.

Comment (kanenathan213) [inline partials/time_to_retirement.html]: Hmm, I tried I few things I thought you meant but can't figure it out. Not sure how that 'value' property update the scope_object everywhere. I definitely want to make this more elegant, so let's talk it through the next time we meet.

File: partials/about.html (hunk 6 above)
Comment (patelspratik) [inline partials/about.html]: I have my small piece with the extra page done, but I'm lukewarm on whether
it's really that useful. I can put it up for review but I've been trying to
think of something better. I haven't obviously haha. Feels...I dunno. I
thought about the extra inputs and stuff too... I don't know, something
doesn't quite jive. Seems like the single boxes is solid.

Tempted to write something around accepting 401ks, but not quite sure. Mr
mustache had an interesting view on when to stop contributing because you
have enough for target, that would be pretty nice.
